src/TLuaInterpreter.cpp

```diff
@@ -11830,35 +11830,288 @@ int TLuaInterpreter::sendSocket( lua_State * L )
+    } else {
+        channels = dlgIRC::readIrcChannels(pHost);
+    }
+
+    lua_newtable(L);
+    int total = channels.count();
+    for (int i = 0; i < total; ++i) {
+        lua_pushnumber(L, i+1);
+        lua_pushstring(L, channels[i].toUtf8().data());
+        lua_settable(L, -3);
+    }
+    return 1;
+}
+
+/** ircGetConnectedHost()
+ *  Returns the Hostname of the connected IRC Server, as provided by the server itself.
+ *  This will return false + error message until the IRC Client has connected and received
+ *  a hostname message from the server.
+ */
+int TLuaInterpreter::ircGetConnectedHost(lua_State* L)
+{
+    Host* pHost = &getHostFromLua(L);
+    QString cHostName = "";
+    QString error = "no client active.";
+    if (mudlet::self()->mpIrcClientMap[pHost]) {
+        cHostName = mudlet::self()->mpIrcClientMap[pHost]->getConnectedHost();
+
+        if (cHostName.isEmpty()) {
+            error = "not yet connected.";
+        }
+    }
+
+    if (cHostName.isEmpty()) {
+        lua_pushboolean(L, false);
+        lua_pushstring(L, error.toUtf8().constData());
+    } else {
+        lua_pushboolean(L, true);
+        lua_pushstring(L, cHostName.toUtf8().constData());
+    }
+    return 1;
+}
+
+/** ircSaveNick( nick )
+ *  Updates IRC client nickname configuration value.
+ *  Does not apply changes to active client until ircRestart() is called.
+ */
+int TLuaInterpreter::ircSaveNick(lua_State* L)
+{
+    string nick;
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "ircSaveNick: bad argument #1 type (nick as string expected, got %s!)", lua_typename(L, 1));
+        return lua_error(L);
+    } else {
+        nick = lua_tostring( L, 1 );
+        if (nick.empty()) {
+            lua_pushnil(L);
+            lua_pushfstring(L, "bad argument #1 value (nick must not be empty!)");
+            return 2;
+        }
+    }
+
+    Host* pHost = &getHostFromLua(L);
+    QPair<bool, QString> result = dlgIRC::writeIrcNickName(pHost, QString::fromStdString(nick));
+    if (!result.first) {
+        lua_pushnil(L);
+        lua_pushfstring(L, "unable to save nick name, reason: %s", result.second.toUtf8().constData());
+        return 2;
+    }
+
+    lua_pushboolean(L, true);
+    return 1;
+}
+
+/** ircSaveServer( hostname, port )
+ *  Updates IRC client connection configuration values with the given hostname and port values.
+ *  Port argument is optional and defaults to 6667.
+ *  Does not apply changes to active client until ircRestart() is called.
+ */
+int TLuaInterpreter::ircSaveServer(lua_State* L)
+{
+    string addr;
+    int port = 6667;
+    if (!lua_isstring(L, 1)) {
+        lua_pushfstring(L, "ircSaveServer: bad argument #1 type (hostname as string expected, got %s!)", lua_typename(L, 1));
+        return lua_error(L);
+    } else {
+        addr = lua_tostring( L, 1 );
+        if (addr.empty()) {
+            lua_pushnil(L);
+            lua_pushfstring(L, "bad argument #1 value (hostname must not be empty!)");
```

Comment:
This is a misuse of the API, not a runtime error though. It should have that bad argument blah style.